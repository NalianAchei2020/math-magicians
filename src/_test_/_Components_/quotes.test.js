import React from 'react';
import { render } from '@testing-library/react';
import { unmountComponentAtNode } from 'react-dom';
import { act } from 'react-dom/test-utils';
import pretty from 'pretty';
import MockAdapter from 'axios-mock-adapter';
import axios from 'axios';

import Qoutes from '../../components/qoutes';

let container = null;
let mock = null;

beforeEach(() => {
  // Setup a DOM element as a render target
  container = document.createElement('div');
  document.body.appendChild(container);

  // Create a new mock adapter instance
  mock = new MockAdapter(axios);
});

afterEach(() => {
  // Cleanup on exiting
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

it('testing for the UI of QuotePage', async () => {
  // Define the mock response data
  const responseData = [
    {
      quote: 'Test quote',
      author: 'Test author',
      category: 'Test category',
    },
  ];

  // Mock the API request
  mock
    .onGet('https://api.api-ninjas.com/v1/quotes?category=happiness')
    .reply(200, responseData);

  await act(async () => {
    render(<Qoutes />, container);
  });

  expect(pretty(container.innerHTML)).toMatchSnapshot();
});
