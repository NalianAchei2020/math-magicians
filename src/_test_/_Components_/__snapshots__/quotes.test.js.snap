// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing for the UI of QuotePage 1`] = `
"<div class=\\"container qoutes-container\\">
  <div>loading....</div>
  <div>Cannot read properties of undefined (reading 'data')</div>
</div>"
`;
